### PR TITLE
Use a precomputed serviceName and method name instead of splitting a full method name on demand

### PR DESCRIPTION
**Pull request by ikhoon**
…full method name on demand

Motivation:

A service name and method name are extracted from a full method name for every request.
It consumes a few CPU cycles. 😆
image

Actually, the service names and their method names are good to cache because they are not changed at all.

Modifications:

- Add `serviceName()` to `RpcRequest` in addition to `serviceType()`
- Add some fatory methods taking a service name to `RpcRequest`.
- Early build simple method names and reuse them for better performance
- Remove `ServiceNamingUtil.fullTypeRpcServiceName()` in favor of `RpcRequest.serviceName()`

Result:

Optimized code and slightly better performance

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/common/RpcRequest.java [hunk 1]
```diff
@@ -64,22 +87,29 @@ static RpcRequest of(Class<?> serviceType, String method, Iterable<?> params) {
...
     }
 
+
     /**
      * Creates a new instance with the specified parameters.
      */
     static RpcRequest of(Class<?> serviceType, String method, Object... params) {
+        return of(serviceType, null, method, params);
+    }
+    /**
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/RpcRequest.java:101]:
```suggestion
    }

    /**
```

File: core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java [hunk 2]
```diff
@@ -1054,14 +1053,13 @@ private void setNamesIfAbsent() {
...
                 } else if (rpcReq != null) {
-                    newServiceName = ServiceNamingUtil.fullTypeRpcServiceName(rpcReq);
+                    newServiceName = rpcReq.serviceName();
                 }
             }
 
             if (newName == null) {
                 if (rpcReq != null) {
                     newName = rpcReq.method();
-                    newName = newName.substring(newName.lastIndexOf('/') + 1);
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java:1064]: 😱

**Review by minwoox — APPROVED** [review]
Thanks!

**Comment by minwoox** [comment]
https://github.com/line/armeria/pull/3537/checks?check_run_id=2542675805#step:7:2005

Could you check the test failure? 😱

File: core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java (hunk 2 above)
Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java:1064]: Actually, it was intended for a gRPC service method. The method for gRPC was a full method name. So we need to cut the service name part from it.

**Comment by trustin** [comment]
Changed the label to 'new feature' because we introduced a new method.

**Pull request merged**